### PR TITLE
Move all TODOs and FIXMEs to GitHub issues

### LINKED ISSUE #63: Move TODO + FIXME from code to here (Issues)
More visibility, higher chance of getting solved.

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
# Description

Closes #63.

## Further considerations

Self-review will show references from each change to a created issue.

- [x] I have performed a self-review of my changes
- [x] I have read and understood the [contributing guidelines](/arizona-framework/arizona/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: priv/static/assets/js/arizona.js [hunk 1]
```diff
@@ -34,11 +34,9 @@ globalThis["arizona"] = (() => {
     const subscribers = new Map()
     const unsubscribers = new Map()
 
-    // @todo Subscribe to global events.
```
Comment (paulo-ferraz-oliveira) [inline priv/static/assets/js/arizona.js:37]: #110

File: priv/static/assets/js/arizona.js [hunk 2]
```diff
@@ -34,11 +34,9 @@ globalThis["arizona"] = (() => {
...
     function subscribe(eventName, callback, opts = {}) {
         let eventSubs = subscribers.get(eventName)
         if (!eventSubs) eventSubs = new Map()
-        // @todo Improve id.
```
Comment (paulo-ferraz-oliveira) [inline priv/static/assets/js/arizona.js:41]: #111

File: src/arizona_html.erl [hunk 3]
```diff
@@ -31,7 +31,6 @@ HTML support.
 %% API funtions.
 %% --------------------------------------------------------------------
 
-% TODO: Do this really safe for HTML.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_html.erl:34]: #112

File: src/arizona_js.erl [hunk 4]
```diff
@@ -42,7 +42,6 @@ send(Event, Payload) ->
 %% Internal funtions.
 %% --------------------------------------------------------------------
 
-% TODO: Real safe JS.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_js.erl:45]: #113

File: src/arizona_live_reload.erl [hunk 5]
```diff
@@ -96,7 +96,6 @@ handle_info({_Pid, {fs, file_event}, {File, Events}},
                 [modified, closed] ->
                     {noreply, State#state{files = Files#{File => {erl, modified}}}};
                 _ ->
-                    % TODO: Act on created, renamed, deleted.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_live_reload.erl:99]: #114

File: src/arizona_tpl_compile.erl [hunk 6]
```diff
@@ -43,7 +43,6 @@ compile({Mod, Fun, Args}) when is_atom(Mod), is_atom(Fun) ->
         directives => #{stateful => true},
         attrs => []}}], [], 0, #state{view = Mod}),
     {ok, Block};
-% TODO: Maybe require a module to be the view.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:46]: #119

File: src/arizona_tpl_compile.erl [hunk 7]
```diff
@@ -74,11 +73,6 @@ compile_tag(#{name := Name} = Tag, Txt, T, P, I, State) ->
 norm_tag_attrs(#{attrs := Attrs0, directives := Dirs}, Id, Target) ->
     maps:fold(fun
         (stateful, true, Attrs) ->
-            % TODO: Rename 'arz-id' to 'arz-sid' (statefull id),
-            %       and provide a 'arz-tid' (tag id) to each tag.
-            %       I think this can improve changes patch,
-            %       because just the most close tag could be rendered
-            %       instead of the entirely block/tree.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:81]: #115

File: src/arizona_tpl_compile.erl [hunk 8]
```diff
@@ -123,8 +117,6 @@ do_compile_tag([K | T], Tag, TT, P, I, State, Acc) ->
     do_compile_tag(T, Tag, TT, P, I, State, [K, $\s | Acc]);
 % IMPORTANT: Text concat must be reviewed when :if and :for be implemented.
 %            Cannot concat when this kind of directive is defined.
-% TODO: There are more optimizations to do by concatenating tags.
-%       Nested blocks of blocks are not concatenated.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:127]: #120

File: src/arizona_tpl_compile.erl [hunk 9]
```diff
@@ -245,7 +237,6 @@ block_struct(Block, P, I, State) ->
     {Mod, Fun} = block_mod_fun(Block, State),
     Tree = apply(Mod, Fun, [maps:get(args, Block, #{})]),
     Attrs = block_attrs(Block),
-    % TODO: Check this directives merge implementation.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:248]: #120

File: src/arizona_tpl_compile.erl [hunk 10]
```diff
@@ -257,8 +248,6 @@ block_struct(Block, P, I, State) ->
     AllAttrs = maps:merge(Attrs, maps:without(NonAttrs, Directives)),
     Id = id(P, I),
     Stateful = maps:get(stateful, Directives, false),
-    % TODO: Remove vars prop from expr tokens.
-    %       They will live in the block props.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:261]: #120

File: src/arizona_tpl_compile.erl [hunk 11]
```diff
@@ -575,10 +562,6 @@ counter(Macros) ->
     {%       and adds this to the blocks param in the state.    }
     <div id={_@id} :stateful>
         <span>
-            {% TODO: Find a better way to define default values }
-            {%       in a pure Erlang implementation.           }
-            {%       Using try/catch by now for this.           }
-            {%       Should find a way to define a maps:get/3.  }
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:581]: #120

File: src/arizona_tpl_compile.erl [hunk 12]
```diff
@@ -285,7 +274,6 @@ block_struct(Block, P, I, State) ->
         vars => vars_group(vars(ExprVars, Id))
     }.
 
-% TODO: Change to binary_to_existing_atom.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:288]: #118

File: src/arizona_tpl_compile.erl [hunk 13]
```diff
@@ -311,7 +299,6 @@ find_first_tag(_) ->
 is_tag(Name) ->
     not lists:member(Name, [<<"!DOCTYPE">>, <<"!doctype">>, <<"?xml">>]).
 
-% TODO: Use binary_to_existing_atom.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:314]: #118

File: src/arizona_tpl_compile.erl [hunk 14]
```diff
@@ -597,9 +580,6 @@ counter(Macros) ->
         {%       It's filled with tokens when is not a void tag }
         {%       and the content between `>...</`, for example  }
         {%       > <div>mycontent</div>                         }
-        {% FIXME: This implementation wont work, because it     }
-        {%        needs to be compiled in tokens.               }
-        {%        The 'merl:tsubst' should be used to fix this. }
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_compile.erl:602]: #116

File: src/arizona_tpl_parse.erl [hunk 15]
```diff
@@ -94,7 +94,6 @@ do_parse_tag([tag_close | T0], Props0, Macros) ->
                 true ->
                     Props = case OpenName =:= <<"head">> of
                         true ->
-                            % TODO: Do not reverse.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_parse.erl:97]: #123 / 1

File: src/arizona_tpl_parse.erl [hunk 16]
```diff
@@ -163,12 +162,10 @@ parse_attr(K, {expr, ExprStr}, Macros) ->
 parse_attr(K, {text, Text}, _Macros) ->
     do_parse_attr(K, {text, Text}).
 
-% TODO: Directive keys to atom using binary_to_existing_atom.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_parse.erl:166]: #118

File: src/arizona_tpl_parse.erl [hunk 17]
```diff
@@ -163,12 +162,10 @@ parse_attr(K, {expr, ExprStr}, Macros) ->
...
 do_parse_attr(<<$:, K/binary>>, {text, <<$:, K/binary>>}) ->
     {directive, {binary_to_atom(K, utf8), true}};
 do_parse_attr(<<$:, K/binary>>, V) ->
     {directive, {binary_to_atom(K, utf8), V}};
-% TODO: Comments support.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_parse.erl:171]: #122

File: src/arizona_tpl_scan.erl [hunk 18]
```diff
@@ -30,24 +30,7 @@ Template scanner.
...
-%               {% Expressions will be compiled into a tree... }
-%               <div>foo</div>
-%               """;
-%           Bar ->
-%               """
-%               {% and should be able to eval variables. }
-%               <div>{Bar}</div>
-%               """
-%       end}
-%       """.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_scan.erl:48]: #17

File: src/arizona_tpl_scan.erl [hunk 19]
```diff
@@ -30,24 +30,7 @@ Template scanner.
...
 string(Str) when is_binary(Str) ->
-    % TODO: Location.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_scan.erl:50]: #123 / 2

File: src/arizona_tpl_scan.erl [hunk 20]
```diff
@@ -120,8 +103,6 @@ scan_tag_name(<<$>, Rest/binary>>, Str, Pos, Len) ->
 scan_tag_name(<<C, Rest/binary>>, Str, Pos, Len)
     when C =:= $\s; C =:= $\r; C =:= $\n ->
     Name = binary_part(Str, Pos, Len),
-    % FIXME: If closing_tag we should go to 'scan'
-    %        instead of 'scan_attr_key'.
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_tpl_scan.erl:124]: #117

File: src/arizona_websocket.erl [hunk 21]
```diff
@@ -71,7 +71,6 @@ websocket_handle({text, Msg}, #{sockets := Sockets} = State) ->
     {Target, Event, Payload} = decode_msg(Msg),
     Socket0 = maps:get(Target, Sockets),
     View = maps:get(view, Socket0),
-    % TODO: {reply, Msg, Socket}
```
Comment (paulo-ferraz-oliveira) [inline src/arizona_websocket.erl:74]: #121

**Review by williamthome — DISMISSED** [review]

File: priv/static/assets/js/arizona.js [hunk 22]
```diff
@@ -35,11 +35,9 @@ globalThis["arizona"] = (() => {
   const subscribers = new Map();
   const unsubscribers = new Map();
 
-  // @todo Subscribe to global events.
```
Comment (paulo-ferraz-oliveira) [inline priv/static/assets/js/arizona.js:38]: #110

File: priv/static/assets/js/arizona.js [hunk 23]
```diff
@@ -35,11 +35,9 @@ globalThis["arizona"] = (() => {
...
   function subscribe(eventName, callback, opts = {}) {
     let eventSubs = subscribers.get(eventName);
     if (!eventSubs) eventSubs = new Map();
-    // @todo Improve id.
```
Comment (paulo-ferraz-oliveira) [inline priv/static/assets/js/arizona.js:42]: #111

**Review by williamthome — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
I'm gonna un-resolve all comments since this is now merge. It'll make it easier to come back here for backtracking what originated an issue.

**Pull request merged**